### DISCUSSION TITLE
how to keep wsa | settings | advanced networking enabled in the presence of vpn connection

### DISCUSSION THREAD

**Discussion by myusrn** [LSPosed/MagiskOnWSALocal#221] [Q&A] [answered]
When i use `& .\adb.exe connect 127.0.0.1:58526` followed by `& .\adb.exe shell am start -n 'com.android.settings/.Settings\$WifiSettingsActivity'` to view my wsa pixel 5 emulated device wifi settings it shows "VirtWifi" if  wsa | settings | advanced networking = off and shows "&lt; my local wifi sid &gt;" if advanced networking = on.

It appears that the wsa | settings | advanced networking = on setting, which gives me non-virtualized wifi adapter access to local wifi is more stable and doesn't requirement me intermittently having to change VirtWifi Dns1 from wsl gateway interface address to 8.8.8.8 to get things working again.

That said it also appears that when wsa networking recognizes a long running vpn connection in my windows host environment it will automatically change wsa | settings | advanced networking = on -> off and i'll to disconnect vpn, restart wsa, in order to get option to turn it back on again.

Any insights on whether or not that is the expected behavior and options for forcing wsa | settings | advanced networking = on to stick OR if should not be doing that in presence of an always on split vpn connection?

**Comment by Howard20181** [comment, marked as answer]
This is expected behavior, and Microsoft's blog post introducing advanced networking also states that coexistence with VPNs is not supported at this time.

**Reply by myusrn** [reply]
Looked at [wsa advanced networking microsoft blog- Search (bing.com)](https://www.bing.com/search?q=wsa%20advanced%20networking%20microsoft%20blog) results and found [Update to Windows Subsystem for Android™ on Windows 11 (July 2022) | Windows Insider Blog](https://blogs.windows.com/windows-insider/2022/07/06/update-to-windows-subsystem-for-android-on-windows-11-july-2022/) hit having `Known Issues | Some VPNs may not work with Advanced Networking. If you use a VPN and find Android apps do not have network connectivity, please disable Advanced Networking in the Windows Subsystem for Android Settings app` details.  

Doesn't say how to interpret case where subsystem itself sees issue with the vpn your using, f5 big-ip client in my case, and automatically disables advanced networking and prompts you to restart. Guess we just wait for a post oct 20, 2022 update notice in [Windows Subsystem for Android Archives | Windows Insider Blog](https://blogs.windows.com/windows-insider/tag/windows-subsystem-for-android/) and see if a resolution to that is provided.
